src/ixdat/readers/biologic.py

```diff
@@ -298,9 +304,168 @@ def process_data_line(self, line):
+
+        Args:
+            path_to_file (Path): The full abs or rel path including the ".mpr" extension
+        """
+        from galvani import BioLogic
+
+        path_to_file = Path(path_to_file or self.path_to_file)
+
+        with open(path_to_file, "rb") as mpr_binary_file:
+            mpr_file = BioLogic.MPRfile(mpr_binary_file)
+
+        self.tstamp = mpr_file.timestamp.timestamp()
+        self.df = pd.DataFrame(mpr_file.data)
+
+        # Build the time series from the dataframe
+        self.data_series_list = series_list_from_dataframe(
+            self.df,
+            time_name=t_str,
+            tstamp=self.tstamp,
+            unit_finding_function=get_column_unit_name,
+        )
+        self.tseries = self.data_series_list[0]
+
+    def series_list_from_mpr_eclabfiles(self, path_to_file=None):
+        """Read a biologic .mpr file
+
+        This makes use of the package `eclabfiles`.
+        See: https://github.com/vetschn/eclabfiles.
+        The dataframe read in by `eclabfiles.to_df()` is stored in the returned
+        measurement `meas` as `meas.reader.df`.
+
+        Args:
+            path_to_file (Path): The full abs or rel path including the ".mpr" extension
+        """
+        import eclabfiles  # not a requirement of ixdat, so we import it here.
+
+        self.df = eclabfiles.to_df(str(path_to_file or self.path_to_file))
+
+        units = self.df.attrs["units"]
+        self.tstamp = self.df.attrs["log"]["posix_timestamp"]
+        self.ec_technique = self.df.attrs["settings"]["technique"]
+
+        # Build the time series from the dataframe
+        self.data_series_list = series_list_from_dataframe(
+            self.df,
+            # the unit-free name of the time column, i.e. "time":
+            time_name=t_str.split("/")[0],
+            tstamp=self.tstamp,
+            # to find a column's unit, look it up in the dictionary `units`:
+            unit_finding_function=units.get,
+        )
+        self.tseries = self.data_series_list[0]
+
+    def _update_aliases_and_ensure_essential_series(self):
+        """A helper function completing the data_series_list for biologic readers."""
+        # First, check if any of the biologic aliases are
+        for data_series in self.data_series_list:
+            for name, alias_list in BIOLOGIC_ALIASES.items():
+
+                # We need to make sure the .mpr (unit-less) names are find-able:
+                name_slash_unit = data_series.name + "/" + data_series.unit_name
+                # In that case, this adds, e.g. "time" to the aliases for essential
+                # series "t", since BIOLOGIC_ALIASES["t"] includes "time/s"
+                if data_series.name in alias_list or name_slash_unit in alias_list:
+                    if name in self.aliases:
+                        self.aliases[name].append(data_series.name)
+                    else:
+                        self.aliases[name] = [data_series.name]
```

Comment:
I think you could leave the `break` there. Once one alias is found, then there is no other right?
Also a request to remove the empty line 441 :smile: